readthedocs/embed/views.py

```diff
@@ -231,101 +265,101 @@ def _get_doc_content(project, version, doc):
 
 def parse_sphinx(content, section, url):
     """Get the embed content for the section."""
-    ret = []
-    headers = []
+    body = content.get('body')
+    toc = content.get('toc')
 
-    if not content or not content.get('body') or not content.get('toc'):
+    if not content or not body or not toc:
         return (None, None, section)
 
-    body = content['body']
-    toc = content['toc']
-    for element in PQ(toc)('a'):
-        headers.append(recurse_while_none(element))
+    headers = [
+        recurse_while_none(element)
+        for element in PQ(toc)('a')
+    ]
 
     if not section and headers:
         # If no section is sent, return the content of the first one
         # TODO: This will always be the full page content,
         # lets do something smarter here
         section = list(headers[0].keys())[0].lower()
 
-    if section:
```

Comment:
Here only this big if was removed and the `query_obj` changed, everything else is the same.